### DISCUSSION TITLE
How to rename/remove a node?

### DISCUSSION THREAD

**Discussion by AleksaDjordjic** [gravitational/teleport#5988] [Q&A] [answered]
I installed teleport following [a tutorial I found online](https://www.youtube.com/watch?v=nk1jfIAL5qE)
Used it for some time, but now I'd like to remove some old nodes and rename existing ones, is this possible?

I've tried to edit the /etc/teleport.yaml and change the node-name inside of there, but didn't figure out how to restart teleport once its running - not even how to shut it down, `service teleport stop` doesn't seem to work and neither teleport nor tctl have commands to stop teleport... Am I missing something obvious?
tctl also doesn't seem to have commands to edit or remove nodes, just list and add

I'm on teleport 5, I think its 5.1.2, just "teleport:5" in the docker-compose

**Comment by webvictim** [comment, marked as answer]
To remove nodes, you can just shut down the Teleport process which is running on them. This might be `sudo systemctl stop teleport`, or even just `kill $(pidof teleport)` to shut down the running process. The nodes will be removed from the output of `tsh ls` and the Teleport web UI after they have not checked in for around 10-15 minutes.

To rename existing nodes, you can edit your `/etc/teleport.yaml` file and set the `nodename`:

```yaml
teleport:
  nodename: name-you-want-goes-here
```

If not set, `nodename` defaults to the `hostname` of the machine.

Once you've changed the `nodename`, you should restart Teleport to see the updates - this might be `sudo systemctl restart teleport`, or in Docker you can theoretically just `kill -HUP $(pidof teleport)`.

**Reply by AleksaDjordjic** [reply]
Thanks, editing seems to work fine.

That is what I did initially, but when trying to start it was saying that the port 3025 was already in use...
Didin't think of killing the process instead of just stopping the service which didn't work for whatever reason

**Reply by webvictim** [reply]
In general, when running inside Docker containers you can't use `service` or `systemctl` to stop processes because the container's init process isn't a fully-fledged service manager like sysvinit or systemd. For Teleport we use [`dumb-init`](https://github.com/Yelp/dumb-init) which is exactly what it says - just a dumb init program that doesn't do very much at all :)

Glad you got things working!

**Reply by AleksaDjordjic** [reply]
The Teleport host, auth, etc. is inside of the docker but node themselves aren't, that's why I was confused
